nsds_lab_to_nwb/components/stimulus/stim_value_extractor.py

```diff
@@ -2,67 +2,116 @@
-
-        return stim_values
-
-    def __parse_command(self, command):
-        ''' return (a_a_a, b.b.b) by parsing string 'a_a_a(b.b.b)' '''
-        res = re.match('(\S+)\((\S+)\)', command)
-        return (res.group(1), res.group(2))
+        if self.stim_type == 'continuous':
+            # this feature is only for discrete stimuli; skip
+            return None
 
+        # for synthetic stimuli
+        if self.stim_values_command is not None:
+            stim_values_command = self.stim_values_command
+            if 'gen_tone_stim_vals' in stim_values_command:
+                # for test_tone_stim
+                stim_values = gen_tone_stim_vals()
+            if 'np.ones' in stim_values_command:
+                # for white noises
+                stim_values = eval(stim_values_command)
+            return stim_values
+
+        # for tone and timit stimuli
+        if 'tone' in self.stim_name:
+            stim_values_path = self._get_stim_values_path()
+            stim_values = tone_stimulus_values(stim_values_path)
+            return stim_values
+        if 'timit' in self.stim_name:
+            stim_values_path = self._get_stim_values_path()
+            stim_values = timit_stimulus_values(stim_values_path)
+            return stim_values
+
+        raise ValueError('Unknown input for stimulus parameterization')
+
+    def _get_stim_values_path(self, path_from_metadata=None):
+        # prioritize path from list_of_stimuli.yaml in this package
+        path_from_los = self.stim_info['parameter_path']
+        if len(path_from_los) > 0:
+            return os.path.join(self.stim_lib_path, path_from_los)
+
+        # if empty path in list_of_stimuli.yaml, use metadata input
+        return os.path.join(self.stim_lib_path, path_from_metadata)
 
 
 def tone_stimulus_values(mat_file_path):
-    ''' adapted from mars.configs.block_directory '''
-    with h5py.File(mat_file_path, 'r') as sio:
-        stim_vals = sio['stimVls'][:].astype(int)
-    stim_vals[0,:] = stim_vals[0,:]+8
+    ''' adapted from mars.configs.block_directory
+
+    Parameters:
+    -----------
+    mat_file_path: full path to a .mat file that contains stim_values.
+
+    Returns:
+    --------
+    stim_vals: a 2D array with two rows along the leading dimension.
+        stim_vals[0, :] are the amplitudes,
+        stim_vals[1, :] are the frequencies of the tones.
+        (should confirm!)
+    '''
+    sio = read_mat_file(mat_file_path)
+    stim_vals = sio['stimVls'][:].astype(int)
+
+    # check dimension
+    shape = stim_vals.shape
+    if not (len(shape) == 2) and (2 in shape):
+        # should be a 2D array, with 2 rows (or 2 columns)
+        raise ValueError('stim_vals dimension mismatch')
+    if shape[1] == 2:
+        stim_vals = stim_vals.T
+
+    # this offset value comes from mars; what is this?
+    # variable naming (amp_offset) was by JHB and could be wrong
+    amp_offset = 8
+    stim_vals[0,:] = stim_vals[0,:] + amp_offset
```

Comment:
This might convert the amplitude values which are something like -8 to 0 which represent change in decibles/10 to a set of 0 to 8 labels.